src/core/net/srp_client.cpp

```diff
@@ -1428,6 +1460,75 @@ void Client::HandleTimer(void)
     }
 }
 
+#if OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE
+
+void Client::EnableAutoStartMode(AutoStartCallback aCallback, void *aContext)
+{
+    mAutoStartCallback = aCallback;
+    mAutoStartContext  = aContext;
+
+    VerifyOrExit(!mAutoStartModeEnabled);
+    mAutoStartModeEnabled = true;
+    CheckAutoStart();
+
+exit:
+    return;
+}
+
+void Client::CheckAutoStart(void)
+{
+    uint16_t                                numServers = 0;
+    NetworkData::Service::SrpServer::Info   selectedServer;
+    NetworkData::Service::SrpServer::Info   server;
+    NetworkData::Service::Manager::Iterator iterator;
+
+    VerifyOrExit(mAutoStartModeEnabled);
+
+    // If the client is not running we check if there is any SRP sever
+    // info in Network Data and select one randomly and then start the
+    // client. If the client is already running with a server that was
+    // selected by the auto-start feature, we verify that the selected
+    // server is still present in the Network Data.
+
+    VerifyOrExit(!IsRunning() || mAutoStartDidSelectServer);
+
+    while (Get<NetworkData::Service::Manager>().GetNextSrpServerInfo(iterator, server) == OT_ERROR_NONE)
+    {
+        numServers++;
+
+        // Choose a server randomly (with uniform distribution) from
+        // the list of servers. As we iterate through server entries,
+        // with probability `1/numServers`, we choose to switch the
+        // current selected server with the new entry. This approach
+        // results in a uniform/same probability of selection among
+        // all server entries.
+
+        if ((numServers == 1) || (Random::NonCrypto::GetUint16InRange(0, numServers) == 0))
+        {
+            selectedServer = server;
+        }
+
+        if (IsRunning() && mAutoStartDidSelectServer && (GetServerAddress() == server.mSockAddr))
```

Comment:
Do we prefer the current server if it's working here? This behavior is not stated in the comments above.

Reply:
I think it's already mentioned in the comment at the top before the `while()` loop in this method.
```
    // If the client is not running we check if there is any SRP sever
    // info in Network Data and select one randomly and then start the
    // client. If the client is already running with a server that was
    // selected by the auto-start feature, we verify that the selected
    // server is still present in the Network Data.
```